library/std/src/io/buffered/bufwriter.rs

```diff
@@ -241,6 +241,16 @@ impl<W: Write> BufWriter<W> {
         &self.buf
     }
 
+    /// Returns a pointer to the whole allocation of the internal buffer.
+    ///
+    /// # Safety
+    ///
+    /// BufWriter must be flushed before the returned memory can be used.
+    /// BufWriter must not be used as long as a reference created from that pointer is alive.
+    pub(in crate::io) fn raw_buffer(&mut self) -> *mut [u8] {
```

Comment:
Why is there a raw pointer here? If it's just to avoid uninitialized memory issues this can return `&mut [MaybeUninit<u8>]`.

Reply:
That would require a `&mut self`, but `BufWriter` remains in (limited) use while the allocation is used as buffer. Only to reach through to the underlying Writer so that the memory isn't touched in the meantime.